metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationZkWriter.java

```diff
@@ -194,23 +195,44 @@ void handleTopicsDelta(Function<Uuid, String> deletedTopicNameResolver, TopicsDe
     }
 
     void handleConfigsSnapshot(ConfigurationsImage configsImage) {
-        Set<ConfigResource> brokersToUpdate = new HashSet<>();
+        Set<ConfigResource> newResources = new HashSet<>(configsImage.resourceData().keySet())
```

Comment:
I think we should throw an exception if we get a ConfigResource whose type isn't BROKER or TOPIC.